web-registry/src/utils/price/computeMedianPrice.ts

```diff
@@ -1,15 +1,32 @@
-type Params = {
-  prices: number[];
-};
+export interface Order {
+  quantity: number;
+  usdPrice: number;
+}
+
+export const calculateMedianPrice = (orders: Order[]): number => {
+  // flatten sell orders into an array of individual prices
+  const prices: number[] = [];
+  for (const order of orders) {
+    // NOTE: the following assumes a maximum precision of 6
+    // for (let i = 0; i < order.quantity; i += 0.000001) {
+    //   prices.push(order.usdPrice);
+    // }
```

Comment:
This was producing a memory issue and `0.01` seems sufficient but let me know what you think.

Reply:
SGTM